### ISSUE TITLE
md-input-container label and placeholder bug

### ISSUE THREAD

**Issue by ilanbiala** [angular/material#1279]
When you have markup like this:
```
<md-input-container>
        <label>Email</label>
        <input type="email" placeholder="johnny.appleseed@gmail.com" required="required" autofocus="autofocus" ng-model="email">
</md-input-container>
```
You end up with 
![screen shot 2015-01-24 at 1 50 08 pm](https://cloud.githubusercontent.com/assets/3695164/5888528/49a88ab0-a3d0-11e4-8c14-804a462fb416.png)
and 
![screen shot 2015-01-24 at 1 50 13 pm](https://cloud.githubusercontent.com/assets/3695164/5888529/4f7d303a-a3d0-11e4-9ba5-146b2ccbfa56.png)
The label covers the placeholder. What should be done about this?

**Comment by ajoslin** [comment]
Thanks. The input's placeholder will now animate in when the input is focused and does not have a value.

**Comment by ajoslin** [comment]
Check out the 'First Name' input: https://material.angularjs.org/#/demo/material.components.input

**Comment by ilanbiala** [comment]
@ajoslin quick fix, nice. It would be awesome to include this in 0.7.1. Are you guys open to docs suggestions?

**Comment by ajoslin** [comment]
More open than you know. @marcysutton is working on a redesign now to make them more usable.

**Comment by marcysutton** [comment]
Totally open to docs suggestions, especially for the API and demos. :+1:

**Comment by ilanbiala** [comment]
@marcysutton for the input docs page, I would make it less of a general page and more like "Here's what the placeholder attribute usage looks like" with a demo and then the code after. Have much more specific  examples helps understand how each piece works.

**Comment by yinhm** [comment]
I have noticed this break the peace with Angular UI Bootstrap.

I will migrate to angular-material in the end, but right now there are features have not been implemented, eg: typeahead.

```
          <form class="navbar-form navbar-right" role="search" ng-controller="SearchController" ng-submit="submit()">
	    <div class="row">
	      <div class="col-lg-12">
		<div class="input-group">
                  <input type="text"
                         class="form-control" 
                         placeholder="Search symbol..."
                         name="q"
                         autocomplete="off"
                         spellcheck="false"
                         ng-model="selected" 
                         typeahead="symbol for symbol in symbols | filter:$viewValue | limitTo:8"
                         typeahead-on-select="submit($item)"
                         />
		  <span class="input-group-btn">
                    <button class="btn btn-default" type="submit">Search</button>
                  </span>
                </div>
              </div>
            </div>
          </form>
```

**Comment by ethanlmovio** [comment]
nice fix. I upgraded to master and this works like a charm.

**Comment by vovafeldman** [comment]
I'm using angular 1.4.4 and angular-material 0.11 and still have the issue. Here's the HTML:

            <md-input-container flex>
                <label>Email address</label>
                <input type="email"
                       ng-model="addresses[p.id]"
                       ng-model-options="{ updateOn: 'blur' }"
                       placeholder="{{ p.email }}" />
            </md-input-container>

That's on focus:
![image](https://cloud.githubusercontent.com/assets/6830121/10028916/3416b822-613d-11e5-9484-9994a0b1767e.png)

That's without focus:
![image](https://cloud.githubusercontent.com/assets/6830121/10028925/3db72b1e-613d-11e5-9a1e-71eec9066ee1.png)

**Comment by jayj** [comment]
@vovafeldman What if you use `md-placeholder` instead of  `placeholder`. Does that change anything?

**Comment by vovafeldman** [comment]
md-placeholder doesn't seem to work at all:
![image](https://cloud.githubusercontent.com/assets/6830121/10034634/43913e6c-6160-11e5-89bd-13b4e0873e2e.png)

**Comment by topherfangio** [comment]
@vovafeldman This should be fixed by #4623. Can you try again with the latest master code?

Also, we updated the [inputs demo](https://material.angularjs.org/HEAD/#/demo/material.components.input) so that the Postal Code input uses both a label and placeholder as an example.

**Comment by vovafeldman** [comment]
Where can I find the concatenated master file?

**Comment by topherfangio** [comment]
*  Angular Material HEAD
 *  Bower Install - https://github.com/angular/bower-material/blob/master/angular-material.js
 *  Bower Install - https://github.com/angular/bower-material/blob/master/angular-material.css 
 *  RAWGIT - https://rawgit.com/angular/bower-material/master/angular-material.js
 *  RAWGIT - https://rawgit.com/angular/bower-material/master/angular-material.css

**Comment by vovafeldman** [comment]
Indeed - it works with the master! I guess the fix isn't merged to 0.11.

**Comment by topherfangio** [comment]
@vovafeldman Indeed, the fix was pretty recent. Should be available in 0.11.1 which should be landing soon.

**Comment by imVinayPandya** [comment]
@topherfangio @kara guys please look at this placeholder issue.

https://github.com/angular/material2/issues/2661#issuecomment-276635417
